irohad/main/application.cpp

```diff
@@ -141,6 +132,21 @@ Irohad::Irohad(const boost::optional<std::string> &block_store_dir,
       })) {
     log_->error("Storage initialization failed: {}", e.value());
   }
+  auto settingsQuery = storage->createSettingQuery();
```

Comment:
I'd move this code to something like `initSettings` which would return Error if it fails and prevent further initialization this way

Reply:
Added